lib/ModuleWarning.js

```diff
@@ -2,15 +2,20 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-function ModuleWarning(module, warning) {
-	Error.call(this);
-	Error.captureStackTrace(this, ModuleWarning);
-	this.name = "ModuleWarning";
-	this.module = module;
-	this.message = warning;
-	this.warning = warning;
+"use strict";
+
+class ModuleWarning extends Error {
+	constructor(module, warning) {
+		super();
+
+		if(Error.hasOwnProperty("captureStackTrace")) {
+			Error.captureStackTrace(this, ModuleWarning);
+		}
+		this.name = "ModuleWarning";
+		this.module = module;
+		this.message = warning;
```

Comment:
I'm just keeping how it was before. I can't change it because I can't track all impacts. That's the reason why I kept all the current behaviors and attributes.